packages/components/simple-table/src/simple-table.story.js

```diff
@@ -126,10 +128,115 @@ const columns = [
   },
 ];
 
+const ColumnConfigForm = props => {
+  const formik = useFormik({
+    initialValues: {
+      width: props.column.width,
+      align: props.column.align,
+      onClick: !!props.column.onClick,
+      isTruncated: !!props.column.isTruncated,
+      shouldIgnoreRowClick: !!props.column.shouldIgnoreRowClick,
+    },
+    onSubmit: values => {
+      console.log(values.shouldIgnoreRowClick);
+
+      const updatedColumn = {
+        ...props.column,
+        width: values.width,
+        align: values.align,
+        onClick: values.onClick ? () => alert('Cell click!') : undefined,
+        isTruncated: values.isTruncated,
+        shouldIgnoreRowClick: values.shouldIgnoreRowClick,
+      };
+      props.updateColumn(updatedColumn);
+    },
+    enableReinitialize: true,
+  });
+
+  return (
+    <div>
+      <h4>{props.column.label}</h4>
+      <form onSubmit={formik.handleSubmit}>
+        <label>
+          width
+          <input
+            name="width"
+            type="text"
+            onChange={formik.handleChange}
+            value={formik.values.width}
+          />
+        </label>
+        <label>
+          align
+          <input
+            name="align"
+            type="text"
+            onChange={formik.handleChange}
+            value={formik.values.align}
+          />
+        </label>
+        <label>
+          onClick
+          <input
+            name="onClick"
+            type="checkbox"
+            onChange={formik.handleChange}
+            checked={formik.values.onClick}
+          />
+        </label>
+        <label>
+          isTruncated
+          <input
+            name="isTruncated"
+            type="checkbox"
+            onChange={formik.handleChange}
+            checked={formik.values.isTruncated}
+          />
+        </label>
+        <label>
+          shouldIgnoreRowClick
+          <input
+            name="shouldIgnoreRowClick"
+            type="checkbox"
+            onChange={formik.handleChange}
+            checked={formik.values.shouldIgnoreRowClick}
+          />
+        </label>
+        <button type="submit" disabled={!formik.dirty}>
+          {'Apply Changes'}
+        </button>
+      </form>
+    </div>
+  );
+};
+ColumnConfigForm.displayName = 'ColumnConfigForm';
+ColumnConfigForm.propTypes = {
+  updateColumn: PropTypes.func.isRequired,
+  column: PropTypes.shape({
+    label: PropTypes.string,
+    width: PropTypes.string,
+    align: PropTypes.string,
+    onClick: PropTypes.func,
+    isTruncated: PropTypes.bool,
+    shouldIgnoreRowClick: PropTypes.bool,
+  }),
+};
+
 storiesOf('Components|Table (NEW)', module)
   .addDecorator(withKnobs)
   // .addDecorator(withReadme(Readme))
   .add('SimpleTable', () => {
+    const [tableData, setTableData] = React.useState({
+      rows: items,
+      columns: initialColumnsState,
+    });
+
+    const handleUpdateColumn = (column, colIndex) => {
+      const newColumns = tableData.columns;
```

Comment:
I'm not sure but I think this assignment is mutable, i.e changes to `newColumns[colIndex]` will also affect `tableData.columns[colIndex]`.

Reply:
You're right! This was a bit half-assed 😳